### PR TITLE
Support modifiers for generated `trait`s

### PR DESCRIPTION
**Pull request by armanbilge**
e.g. `private[package]`, `sealed`, etc.

### CODE REVIEW

File: shared/src/main/scala/com/raquo/domtypes/codegen/generators/PropsTraitGenerator.scala [hunk 1]
```diff
@@ -10,6 +10,8 @@ class PropsTraitGenerator(
   override protected val headerLines: List[String],
   override protected val traitCommentLines: List[String],
   override protected val traitName: String,
+  override protected val traitModifiers: List[String],
+
```
Comment (armanbilge) [inline shared/src/main/scala/com/raquo/domtypes/codegen/generators/PropsTraitGenerator.scala:14]:
```suggestion
  override protected val traitModifiers: List[String],
```

File: shared/src/main/scala/com/raquo/domtypes/codegen/generators/EventPropsTraitGenerator.scala [hunk 2]
```diff
@@ -8,6 +8,8 @@ class EventPropsTraitGenerator(
   override protected val defGroupComments: EventPropDef => List[String],
   override protected val headerLines: List[String],
   override protected val traitCommentLines: List[String],
+  override protected val traitModifiers: List[String],
+
```
Comment (armanbilge) [inline shared/src/main/scala/com/raquo/domtypes/codegen/generators/EventPropsTraitGenerator.scala:12]:
```suggestion
  override protected val traitModifiers: List[String],
```

**Comment by armanbilge** [comment]
CI is a bit flaky, seems like there is some sort of race condition.

**Comment by raquo** [comment]
Thanks!

CompileSpec requires the files generated by GeneratorSpec. The order of platforms is set correctly when running tests from the root project, but I think the problem is that `writeToFile` does not wait for the file to be written to disk before returning (apparently flushing the stream is not enough?). I've addressed that, let's hope this fixes it.

**Pull request merged**